lib/test_formulae.rb

```diff
@@ -13,8 +13,12 @@ def initialize(tap:, git:, dry_run:, fail_fast:, verbose:)
 
       protected
 
-      def bottled?(formula, tag = nil, no_older_versions: false)
-        formula.bottle_specification.tag?(Utils::Bottles.tag(tag), no_older_versions: no_older_versions)
+      def bottled?(formula, no_older_versions: false)
+        if formula.bottle_specification.tag?(Utils::Bottles.tag(:all))
+          formula.deps.all? { |dep| bottled?(dep.to_formula, no_older_versions: no_older_versions) }
```

Comment:
Should this be recursive deps? Feels like a comment explaining why this is done would be useful.

Reply:
No need. This procedure guarantees that the recursive dependencies are bottled too because a formula won't be bottled unless its dependencies are. (The handful that were accidentally bottled during mass bottling have been bumped since.)

We only need to recurse further up the tree whenever we find an `:all` bottle.

Reply:
Gotcha.

Reply:
> This procedure guarantees that the recursive dependencies are bottled too because a formula won't be bottled unless its dependencies are. (The handful that were accidentally bottled during mass bottling have been bumped since.)

This may be the case in core, but it is not the case in `osrf/simulation` right now. There are some formulae with `big_sur` bottles that were built before Nov 3 against `catalina` or earlier bottles of at least one dependency. Now, when rebuilding these formulae, they lose their `big_sur` bottle since their dependency chain isn't completely bottled on `big_sur` (for example https://github.com/osrf/homebrew-simulation/commit/140ece04831a4057629f5e1698405caa28a221f9).

I'll try to test and find a reproducible test case from the `osrf/simulation` tap. Thanks for you help with all this!

Reply:
Changing `deps` to `recursive_dependencies` here only makes your problem worse, not better, though.

Reply:
fair enough; I don't fully understand all this code. thanks for your patience

we don't yet have `:all` bottles; we need to resolve https://github.com/osrf/homebrew-simulation/issues/1682 before we will get `:all` bottles automatically from `test-bot`